### ISSUE TITLE
Forms in Woven are annoying

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#11256]
|[<img alt="radix's avatar" src="https://avatars.githubusercontent.com/u/227068?s=50" width="50" height="50">](https://github.com/radix)| @radix reported|
|-|-|
|Trac ID|trac#73|
|Type|enhancement|
|Created|2003-07-14 17:39:41Z|

<details><summary>Searchable metadata</summary>

```
trac-id__73 73
type__enhancement enhancement
reporter__radix radix
priority__low low
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__ 
keywords__web web
time__1058204381000000 1058204381000000
changetime__1464994060561329 1464994060561329
version__ 
owner__ 
cc__glyph cc__radix cc__dp
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="radix's avatar" src="https://avatars.githubusercontent.com/u/227068?s=50" width="50" height="50">](https://github.com/radix)|@radix commented|
|-|-|

```
#!html
<pre>
Currently, when you submit data through a Woven Form that
raises a FormError for one reason or another, another form
is generated for you in a plain white page.

It is preferable to have the error reported in-line with the
_original_ page that displayed the form. This is tricky, but
there are ways to do it. Here's one (so far sketchy) idea.

I propose a "FormPage" class that allows the programmer to
specify one or more Forms that will be used in a Page. All
of these forms, in the template, should have an "action"
that references the same FormPage. The MethodSignatures for
these forms will be associated with a callback (as usual)
and a Redirect (as idiomatically usual) that points to,
e.g., a results display page.

So, our FormPage is also the processor of all forms used in
that FormPage. If a FormError occurs any time during
processing, then the FormPage can rerender itself entirely,
putting an error message near the appropriate form that the
user put bad data into.

Thoughts?

</pre>
```

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)|@glyph commented|
|-|-|

```
#!html
<pre>
We need a FormPage thing, that's for sure, but the one you
suggest is impossible.  a URL for GET may specify multiple
forms to display, but a URL for POST must only specify one
form to submit.

The FormPage we need is a Resource that has *2* URLs for *1*
form.  That way you can bounce back and forth between
displaying errors on the "display" template without creating
confusion on the "post" URL

take a look in sandbox/glyph/formdemo for an implementation

</pre>
```

**Comment by twisted-trac** [comment]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)|@itamarst commented|
|-|-|

```
#!html
<pre>
Looks like dp is on to this.

</pre>
```

**Comment by twisted-trac** [comment]
|[<img alt="jyknight's avatar" src="https://avatars.githubusercontent.com/u/873903?s=50" width="50" height="50">](https://github.com/jyknight)|@jyknight commented|
|-|-|

```
#!html
<pre>
This bug is being resolved, wontfix, as woven is "almost deprecated" and unmaintained. Use nevow 
instead. See &lt;http://divmod.org/users/dp.twistd/wiki/> and the twisted-web mailing list &lt;http://
twistedmatrix.com/cgi-bin/mailman/listinfo/twisted-web>.
</pre>
```

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="cyli's avatar" src="https://avatars.githubusercontent.com/u/223200?s=50" width="50" height="50">](https://github.com/cyli)<a name="note_6"></a>|@cyli commented|
|-|-|

In changeset e56d76fad98d153b4cab70560b0233c93578dce5
```
#!CommitTicketReference repository="" revision="e56d76fad98d153b4cab70560b0233c93578dce5"
Merge pull request #73 from twisted/7229-rodrigc-travis: add travis configuration to cover some buildbot tests

Author: adiroiban, rodrigc
Reviewer: hawkowl, adiroiban, mzdaniel, cyli
Fixes: #7229

Sets up a travis configuration to cover linux py27, py33, py34, py35 tests, and updates some of the Github contribution documentation to reference the travis results.
```
